### PR TITLE
Clarify language around unlisted and direct posts

### PR DESCRIPTION
**Pull request by trwnh**
- Renames "Unlisted" to "Not in public timelines"
- Emphasize "mentioned people only" applies to the current post

### CODE REVIEW

File: app/javascript/mastodon/locales/defaultMessages.json [hunk 1]
```diff
@@ -726,7 +726,7 @@
         "id": "privacy.public.short"
       },
       {
-        "defaultMessage": "Unlisted",
```
Comment (ykzts) [inline app/javascript/mastodon/locales/defaultMessages.json:729]: https://github.com/mastodon/mastodon/blob/fefda70ee063b1c53b2bbbdbfcfc7f043c2b3b4d/app/javascript/mastodon/components/status.js#L56

File: app/javascript/mastodon/locales/defaultMessages.json [hunk 2]
```diff
@@ -1570,27 +1570,27 @@
         "id": "privacy.public.long"
       },
       {
-        "defaultMessage": "Unlisted",
```
Comment (ykzts) [inline app/javascript/mastodon/locales/defaultMessages.json:1573]: https://github.com/mastodon/mastodon/blob/fefda70ee063b1c53b2bbbdbfcfc7f043c2b3b4d/app/javascript/mastodon/features/compose/components/privacy_dropdown.js#L15

File: app/javascript/mastodon/locales/defaultMessages.json [hunk 3]
```diff
@@ -1570,27 +1570,27 @@
...
+        "defaultMessage": "Not in public timelines",
         "id": "privacy.unlisted.short"
       },
       {
-        "defaultMessage": "Visible for all, but opted-out of discovery features",
```
Comment (ykzts) [inline app/javascript/mastodon/locales/defaultMessages.json:1577]: https://github.com/mastodon/mastodon/blob/fefda70ee063b1c53b2bbbdbfcfc7f043c2b3b4d/app/javascript/mastodon/features/compose/components/privacy_dropdown.js#L16

File: app/javascript/mastodon/locales/defaultMessages.json [hunk 4]
```diff
@@ -1570,27 +1570,27 @@
...
+        "defaultMessage": "Visible for all, but not in public and hashtag timelines.",
         "id": "privacy.unlisted.long"
       },
       {
         "defaultMessage": "Followers only",
         "id": "privacy.private.short"
       },
       {
-        "defaultMessage": "Visible for followers only",
```
Comment (ykzts) [inline app/javascript/mastodon/locales/defaultMessages.json:1585]: https://github.com/mastodon/mastodon/blob/fefda70ee063b1c53b2bbbdbfcfc7f043c2b3b4d/app/javascript/mastodon/features/compose/components/privacy_dropdown.js#LL18

File: app/javascript/mastodon/locales/defaultMessages.json [hunk 5]
```diff
@@ -1570,27 +1570,27 @@
...
+        "defaultMessage": "Visible to anyone following you",
         "id": "privacy.private.long"
       },
       {
         "defaultMessage": "Mentioned people only",
         "id": "privacy.direct.short"
       },
       {
-        "defaultMessage": "Visible for mentioned users only",
```
Comment (ykzts) [inline app/javascript/mastodon/locales/defaultMessages.json:1593]: https://github.com/mastodon/mastodon/blob/fefda70ee063b1c53b2bbbdbfcfc7f043c2b3b4d/app/javascript/mastodon/features/compose/components/privacy_dropdown.js#L20

**Review by ykzts — CHANGES_REQUESTED** [review]
If the origin file is not changed, it will rollback with `yarn manage:translations`.

**Comment by trwnh** [comment]
@ykzts Changed all the components and source files corresponding to the changed messages, thank you ^_^;

**Review by ykzts — APPROVED** [review]

**Comment by ClearlyClaire** [comment]
Unlisted also opts out of trending

**Review by Gargron — COMMENTED** [review]
Touching the descriptions just highlights how much "unlisted" doesn't fit in with the other options. There's no good way to summarize what it does. "Public with caveats". "Same as public, but won't appear in a hashtag timeline (if you use a hashtag) and won't appear in trends (if it becomes popular and you opted-in to being discoverable in a different place). It also won't appear in the local and federated timelines (but it won't appear anyway if it's a reply to someone, and it might still appear in a different implementation like Pleroma because "unlisted" isn't *really* an ActivityPub thing because local and federated timelines are not an ActivityPub thing at all)."

Other description changes look fine.

**Comment by trwnh** [comment]
I think in the long term "unlisted" should be replaced by `toot:discoverable` flag being applied to statuses and not just profiles, and maybe making it a separate per-post toggle for discoverability, but so long as we have current privacy scopes, maybe calling it "Semi-Public" isn't so bad. In any case, we can't really control what other implementations do, and it doesn't feel worth mentioning non-self-reply-filtering from timelines either, as that could change. So what about the following?

"Semi-Public"

"Visible to anyone on your profile and in home timelines[, but not in public timelines, hashtags, or trends]"

I'm not sure if the second half is strictly necessary or implied by the first half, or if it could be replaced by the word "only". I'm inclined to think it might be. We already have a hint banner that appears when someone tries to use a hashtag in an unlisted post.

**Comment by mjankowski** [comment]
Closing due to no activity in ~1 year, merge conflicts, and lack of consensus around the "unlisted" specific changes.

If you want to push the rest forward, can you rebase and reopen with just the non-unlisted-related changes in this PR?

**Pull request closed without merging**